### PR TITLE
View Angles (Part 1 - Redux)

### PR DESCRIPTION
**Pull request by MajorCooke**
Added ViewAngle/Pitch/Roll properties to actors.

- These are offsets for camera angles that allow turning the camera without affecting aim or movement direction.
- Added A_SetView<Angle/Pitch/Roll>, which will set the view direction.
- Added ABSVIEWANGLES flag, used to make the view absolute instead of an offset.

(The last PR had some serious issues when another PR was merged so this had to be remade.)

### CODE REVIEW

**Comment by MajorCooke** [comment]
Can I request for reviews here?

I understand that this is adding another set of variables to actor, but honestly, the ability to change cameras prevents me from restricting this to just playerinfo. It's impossible to record offsets that way.

File: src/playsim/p_mobj.cpp [hunk 1]
```diff
@@ -3351,52 +3352,106 @@ DEFINE_ACTION_FUNCTION(AActor, SetShade)
...
+		if (player != nullptr && (fflags & SPF_INTERPOLATE))
+		{
+			player->cheats |= CF_INTERPVIEW;
+		}
+	}
+}
+
+void AActor::SetViewPitch(DAngle p, int fflags)
+{
+	if (player != NULL || (fflags & SPF_FORCECLAMP))
```
Comment (nashmuhandes) [inline src/playsim/p_mobj.cpp:3420]: @MajorCooke just a trivial styling issue but ``NULL`` should probably be changed to ``nullptr`` here

**Comment by MajorCooke** [comment]
I already did that for you, didn't I?

**Comment by Doom2fan** [comment]
> @Doom2fan I already did that for you, didn't I? `A_SetView<Angle/Pitch/Roll>`

There's no direct-native versions in the PR. You can take a look at [this](https://github.com/coelckers/gzdoom/blob/4fb6b7c7d41f015e41b75bac896c1f17c71aa124/src/scripting/vmthunks_actors.cpp#L136) as an example of a function with a direct-native version. These are used by the JIT to turn the function calls into fully native ones.

**Comment by MajorCooke** [comment]
Oh okay. So I just change `DEFINE_ACTION_FUNCTION` to `DEFINE_ACTION_FUNCTION_NATIVE`? Simple enough.

**Comment by Doom2fan** [comment]
> 
> 
> Oh okay. So I just change `DEFINE_ACTION_FUNCTION` to `DEFINE_ACTION_FUNCTION_NATIVE`? Simple enough.

You also have to provide the actual direct-native version of the function.

**Comment by MajorCooke** [comment]
All done, I hope.

**Comment by MajorCooke** [comment]
No one cares if the VS 32-bit for windows fails since 4.3.3 is the last version to support 32-bit, right?

**Comment by coelckers** [comment]
In what way does it fail? If it is because something needs 64 bits, I don't care much, but if it fails because of some programming error, that isn't an excuse not to fix it.

**Comment by MajorCooke** [comment]
Actually all checks now passed. Strange, but okay.

At first I received an email stating https://puu.sh/G2PS0/bd3b109e38.png this one had failed but now it succeeded, so nevermind.

**Comment by Doom2fan** [comment]
Code looks fine to me, and the small attempt at testing-by-fuckery I threw at it* didn't seem to break it.
* (changing pitch + viewpitch simultaneously, in opposite directions)

**Comment by MajorCooke** [comment]
Awesome. So this is ready for merging? I'm pumped. Finally I can fix a few broken systems with angle changing when this gets in and no longer have my forward/side movement screwing up.

**Pull request merged**